packages/itwin/grouping-mapping-widget/src/widget/components/GroupPropertyUtils.ts

```diff
@@ -0,0 +1,364 @@
+      }
+      // Get structs
+      case PropertyValueFormat.Struct: {
+        const nestedContentField = property as NestedContentField;
+        // Only handling single path and not handling nested content fields within navigations
+        if (
+          nestedContentField.pathToPrimaryClass &&
+          nestedContentField.pathToPrimaryClass.length > 1
+        ) {
+          break;
+        }
+
+        switch (nestedContentField.relationshipMeaning) {
+          // Aspects are not looked at again. Extraction does not support aspects within navigations.
+          // Navigation properties
+          case RelationshipMeaning.RelatedInstance: {
+            if (
+              // Deal with a TypeDefinition
+              nestedContentField.pathToPrimaryClass[0].relationshipInfo.name ===
+              "BisCore:GeometricElement3dHasTypeDefinition"
+            ) {
+              return extractNested([...propertyTraversal, "TypeDefinition"], nestedContentField.nestedFields);
+            }
+            break;
+          }
+          default: {
+            // Some elements don't have a path to primary class or relationship meaning..
+            // Most likely a simple struct property
+            if (!nestedContentField.pathToPrimaryClass) {
+              return extractStruct(property);
+            }
+          }
+        }
+      }
+    }
+    return [];
+  });
+
+export const convertPresentationFields = (propertyFields: Field[]): PropertyMetaData[] =>
+  propertyFields.flatMap((property) => {
+    // Generate base ECProperty
+    switch (property.type.valueFormat) {
+      case PropertyValueFormat.Primitive: {
+        const extractedPrimitives = extractPrimitives([], property as PropertiesField);
+        for (const extractedPrimitive of extractedPrimitives) {
+          extractedPrimitive.schema = "*";
+          extractedPrimitive.className = "*";
+          return extractedPrimitive;
+        }
+        break;
+      }
+      // Get structs
+      case PropertyValueFormat.Struct: {
+        const nestedContentField = property as NestedContentField;
+        // Only handling single path and not handling nested content fields within navigations
+        if (
+          nestedContentField.pathToPrimaryClass &&
+          nestedContentField.pathToPrimaryClass.length > 1
+        ) {
+          break;
+        }
+        switch (nestedContentField.relationshipMeaning) {
+          case RelationshipMeaning.SameInstance: {
+            // Check for aspects.
+            if (
+              (nestedContentField.pathToPrimaryClass[0].relationshipInfo
+                .name === "BisCore:ElementOwnsUniqueAspect" ||
+                nestedContentField.pathToPrimaryClass[0].relationshipInfo
+                  .name === "BisCore:ElementOwnsMultiAspects")
+            ) {
+              const fullClassName = nestedContentField.contentClassInfo.name;
+              const schema = fullClassName.split(":")[0];
```

Comment:
Is the `schema` and `class` in `fullClassName` always separated by `:`, or can it be also separated by `.`?

Reply:
Always `:`